api/soap/mc_enum_api.php

```diff
@@ -290,11 +290,25 @@ function mci_enum_get_array_by_id( $p_enum_id, $p_enum_type, $p_lang ) {
 		$t_result['name'] = MantisEnum::getLocalizedLabel( $t_enum_string_value, $t_enum_localized_value, $p_enum_id );
 	} else {
 		$t_enum_array = MantisEnum::getAssocArrayIndexedByValues( $t_enum_string_value );
+
+		if (!array_key_exists($p_enum_id, $t_enum_array)) {
+				log_event( LOG_WEBSERVICE, "Id '$p_enum_id' does not exist in the '$p_enum_type' enumeration"  );
```

Comment:
Not sure about logging here. While the error does occur in the webservice's context, this is really a system configuration problem, not an issue of the web service itself.